### PR TITLE
Allow Apache integration specify the server's trusted CA

### PR DESCRIPTION
**Pull request by mariomac**
#### Description of the changes

Usually the host should be configured so it trusts its own Apache
Certificate Authority, even if it is self-created and not trusted
outside the host.

In case the user don't want to do it, two configuration options have
been provided to allow the user to specify the CA certificate or
the folder where the certificate is.

### Author

- [x] add a risk label after carefully considering the "blast radius" of your changes
- [ ] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [x] assign at least one reviewer

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] ask questions about anything that isn't clear and obvious
- [ ] approve the PR when you consider it's good to merge

### CODE REVIEW

File: integrations/apache/src/apache.go [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 	sdk_args "github.com/newrelic/infra-integrations-sdk/args"
 	"github.com/newrelic/infra-integrations-sdk/log"
 	"github.com/newrelic/infra-integrations-sdk/sdk"
 )
 
 type argumentList struct {
 	sdk_args.DefaultArgumentList
-	StatusURL string `default:"http://127.0.0.1/server-status?auto" help:"Apache status-server URL."`
+	StatusURL    string `default:"http://127.0.0.1/server-status?auto" help:"Apache status-server URL."`
+	CABundleFile string `default:"" help:"Alternative Certificate Authority bundle file"`
```
Comment (MagdaPaj) [inline integrations/apache/src/apache.go]: I'm not sure if this `default:""` it's needed. What happen if you omit `default`?

File: integrations/apache/apache-config.yml.sample [hunk 2]
```diff
@@ -5,6 +5,7 @@ instances:
       command: metrics
       arguments:
           status_url: http://127.0.0.1/server-status?auto
+          ca_cert_dir: /etc/ssl/certs
```
Comment (MagdaPaj) [inline integrations/apache/apache-config.yml.sample]: It should be renamed, right?

File: integrations/apache/src/apache.go [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+	CABundleDir  string `default:"" help:"Alternative Certificate Authority bundle directory"`
 }
 
 const (
 	integrationName    = "com.newrelic.apache"
-	integrationVersion = "1.0.0"
+	integrationVersion = "1.1.0"
+
+	defaultHTTPTimeout = time.Second * 1
```
Comment (MagdaPaj) [inline integrations/apache/src/apache.go]: maybe we could include the `timeout` flag too? But then the problem of testing it. Not sure. The final idea is to also have `timeout` as configurable.
/cc @areina

File: integrations/apache/README.md [hunk 4]
```diff
@@ -19,6 +19,27 @@ In order to use the Apache Integration it is required to configure
...
+
+You can change this URL by modifying the `status_url` instance parameter of your `apache-config.yml` file.
+
+If the `/server-status` is reachable only though an HTTPS connection which uses a certificate that is signed by your own
+Certificate Authority (e.g. for internal use or development purposes), you will need to make sure one of the next actions
+have been taken:
+
+* Install your Certificate Authority in your host. The Apache integration will look for it by default in the host's root
+  bundle for Certificate Authorities.
+* Update the `apache-config.yml` to have at leastone of the next instance parameters configured: `ca_bundle_file` or
```
Comment (MagdaPaj) [inline integrations/apache/README.md]: leastone -> least one

File: integrations/apache/CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## 1.1.0 (2018-02-08)
+### Added
+- Allow working with own's Apache Certificate Authority through the `ca_cert_file` and `ca_cert_dir` configuration
```
Comment (MagdaPaj) [inline integrations/apache/CHANGELOG.md]: renaming required?

File: integrations/apache/README.md [hunk 6]
```diff
@@ -19,6 +19,27 @@ In order to use the Apache Integration it is required to configure
...
+  `ca_bundle_dir`, whose values must be, respectively, the absolute paths to your alternative Certificate Authorities' bundle
+  file (or the directory where they are).
+* Add to the [Infrastructure Agent configuration file](https://docs.newrelic.com/docs/infrastructure/new-relic-infrastructure/configuration/configure-infrastructure-agent))
```
Comment (MagdaPaj) [inline integrations/apache/README.md]: Maybe I misunderstand sth. Why this step is needed, why we need to pass this variables to the agent? It's not enough to have them just configured in our config file?

Comment (mariomac) [reply integrations/apache/README.md]: You just need only one of the three actions. I have reworded it to make it clear that, if you do one of them, don't need to do the rest.

File: integrations/apache/src/apache.go (hunk 1 above)
Comment (mariomac) [reply integrations/apache/src/apache.go]: Good catch. Just removed it and works.

File: integrations/apache/src/apache.go (hunk 3 above)
Comment (mariomac) [reply integrations/apache/src/apache.go]: I just left it as hardcoded to be consistent with the previous implementation and don't loose the actual focus of the task.

**Pull request merged**